server/sonar-db-dao/src/main/java/org/sonar/db/property/InternalPropertiesDao.java

```diff
@@ -77,13 +90,57 @@ public void saveAsEmpty(DbSession dbSession, String key) {
   }
 
   /**
+   * @return a Map with an {link Optional<String>} for each String in {@code keys}.
+   */
+  public Map<String, Optional<String>> selectByKeys(DbSession dbSession, Set<String> keys) {
+    if (keys == null || keys.isEmpty()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)